src/app/CommandHandlerImpl.cpp

```diff
@@ -475,25 +475,6 @@ Status CommandHandlerImpl::ProcessGroupCommandDataIB(CommandDataIB::Parser & aCo
     }
     VerifyOrReturnError(err == CHIP_NO_ERROR, Status::Failure);
 
-    // Per spec, we do the "is this a timed command?" check for every path, but
-    // since all paths that fail it just get silently discarded we can do it
-    // once up front and discard all the paths at once.  Ordering with respect
-    // to ACL and command presence checks does not matter, because the behavior
-    // is the same for all of them: ignore the path.
-#if !CHIP_CONFIG_USE_DATA_MODEL_INTERFACE
-
-    // Without data model interface, we can query individual commands.
-    // Data model interface queries commands by a full path so we need endpointID as well.
-    //
-    // Since this is a performance update and group commands are never timed,
-    // missing this should not be that noticeable.
-    if (CommandNeedsTimedInvoke(clusterId, commandId))
```

Comment:
So in the CHIP_CONFIG_USE_DATA_MODEL_INTERFACE case (the only case now): where is this check done?

Reply:
This was an optimization that was possible with direct ember calls because the command flags were global. In the data model interface we need the full path, so we cannot early-optimize (unless we do more work).

The individual check (that matches spec text more closely) was added in `mpCallback->ValidateCommandCanBeDispatched` (in #35897). Specifically timed invoke validation is done in `CheckCommandFlags`: https://github.com/project-chip/connectedhomeip/blob/master/src/app/InteractionModelEngine.cpp#L1859C61-L1859C78

Reply:
Great, "ValidateCommandCanBeDispatched call below" is perfect.